### PR TITLE
[Windows] Fix DatePicker TextColor property

### LINKED ISSUE #7924: TextColor does not work correctly in Windows for the Picker and DatePicker.
### Description

The text color always remains black even if you put it in a different color. For the picker, the color changes to black when you move the mouse over it, sometimes the color changes just like that. Works correctly in Android.

Another bug is that the calendar icon shown on the DatePicker in Windows is not visible in Android.

I find the calendar very basic. It should be possible to enter the date with the keyboard as well. Now it is almost impossible to enter dates of birth.

### Steps to Reproduce

1. Open the attached app.
2. Try out the Picker and DatePicker.
3. 
[MauiAppTestPickers.zip](https://github.com/dotnet/maui/files/8877727/MauiAppTestPickers.zip)

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 11

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix TextColor property on Windows DatePicker.

image

### Issues Fixed

Fixes #7924

### CODE REVIEW

File: src/Core/src/Platform/Windows/DatePickerExtensions.cs [hunk 1]
```diff
@@ -40,8 +41,28 @@ public static void UpdateCharacterSpacing(this CalendarDatePicker platformDatePi
...
+			else
+			{
+				platformDatePicker.Resources.SetValueForAllKey(TextColorResourceKeys, platformBrush);
+				platformDatePicker.Foreground = platformBrush;
+			}
 		}
+
+		// ResourceKeys controlling the foreground color of the CalendarDatePicker.
+		// https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.calendardatepicker?view=windows-app-sdk-1.1
+		static readonly string[] TextColorResourceKeys =
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/Windows/DatePickerExtensions.cs:61]: This is the main change.

**Comment by jsuarezruiz** [comment]
> Is it possible do add a windows device test for this ?

Sure thing. Added all the existing DatePicker tests to Windows.

windows-datepicker

File: src/Core/src/Platform/Windows/DateTimeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace Microsoft.Maui.Platform
+{
+	// TODO NET7 MAKE PUBLIC
+	internal static class DateTimeExtensions
+	{
+		public static DateTimeOffset ToDateTimeOffset(this DateTime dateTime)
```
Comment (PureWeen) [inline src/Core/src/Platform/Windows/DateTimeExtensions.cs:8]: What's the use case here? Can you add a test for this change?

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Comment by 1888games** [comment]
Should this have fixed the picker control, also detailed in the original issue? I am still seeing black text regardless of what is set, when using the picker control in 17.4.0 Preview 1.

https://github.com/dotnet/maui/issues/7924

**Comment by handysharp** [comment]
Picker text color still not working properly as per 1888games (Sept 8). When over picker, the text goes to black. Sometimes stays like that and sometimes goes back to selected color override in xaml when item selected. I assume it should turn black only when hovering over but otherwise should be the coded color??

**Comment by m-calvi** [comment]
I still have this problem on the DatePicker control, changing the text colour and hovering the mouse over it turns the text black. I'm using .NET 7.0

**Comment by larsduewel** [comment]
Same problem here the PointerOver TextColor can be set, but it is ignored.

**Pull request merged**